source/sessions/tests/unified/snapshot-sessions.json

```diff
@@ -0,0 +1,698 @@
+{
+  "description": "snapshot-sessions",
+  "schemaVersion": "1.0",
+  "runOnRequirements": [
+    {
+      "minServerVersion": "5.0",
+      "topologies": [
+        "sharded-replicaset"
+      ]
+    }
+  ],
+  "createEntities": [
+    {
+      "client": {
+        "id": "client0",
+        "observeEvents": [
+          "commandStartedEvent"
+        ]
+      }
+    },
+    {
+      "database": {
+        "id": "database0",
+        "client": "client0",
+        "databaseName": "database0"
+      }
+    },
+    {
+      "collection": {
+        "id": "collection0",
+        "database": "database0",
+        "collectionName": "collection0",
+        "collectionOptions": {
+          "readPreference": {
+            "mode": "secondary"
+          }
+        }
+      }
+    },
+    {
+      "session": {
+        "id": "session_find0",
+        "client": "client0",
+        "sessionOptions": {
+          "isSnapshot": true
+        }
+      }
+    },
+    {
+      "session": {
+        "id": "session_find1",
+        "client": "client0",
+        "sessionOptions": {
+          "isSnapshot": true
+        }
+      }
+    },
+    {
+      "session": {
+        "id": "session_distinct0",
+        "client": "client0",
+        "sessionOptions": {
+          "isSnapshot": true
+        }
+      }
+    },
+    {
+      "session": {
+        "id": "session_distinct1",
+        "client": "client0",
+        "sessionOptions": {
+          "isSnapshot": true
+        }
+      }
+    },
+    {
+      "session": {
+        "id": "session_aggregate0",
+        "client": "client0",
+        "sessionOptions": {
+          "isSnapshot": true
+        }
+      }
+    },
+    {
+      "session": {
+        "id": "session_aggregate1",
+        "client": "client0",
+        "sessionOptions": {
+          "isSnapshot": true
+        }
+      }
+    },
+    {
+      "session": {
+        "id": "session_mixed",
+        "client": "client0",
+        "sessionOptions": {
+          "isSnapshot": true
+        }
+      }
+    },
+    {
+      "session": {
+        "id": "session_first_read",
+        "client": "client0",
+        "sessionOptions": {
+          "isSnapshot": true
+        }
+      }
+    },
+    {
+      "session": {
+        "id": "session_invalid_snapshot",
+        "client": "client0",
+        "sessionOptions": {
+          "isSnapshot": true,
+          "snapshotTimestamp": 0
```

Comment:
I'm not clear what your intention is here.  In the spec this option is not defined.

Reply:
[I've marked this test as optional](https://github.com/mongodb/specifications/pull/1022/files#diff-6c9d82f6b269142ff8fe9f0ef8d7c90f9e2761c947b5caf92e6820124b81fc2bR299-R300), for drivers that can alter `snapshotTimestamp` field internally.

Not sure what's the best approach here: Mandate drivers to provide functionality to set `snapshotTimestamp` or extract this test as optional prose test.